app/src/ingestion/pdf_postprocess.py

```diff
@@ -223,6 +223,59 @@ def _should_merge_text_split_across_pages(text: EnrichedText, next_text: Enriche
     return next_text.text.strip()[0].islower()
 
 
+def _find_heading_index(heading_list: list[str], heading_text: str) -> int:
+    try:
+        return heading_list.index(heading_text)
+    except ValueError:
+        return -1
+
+
+def _group_headings_text(markdown_texts: list[EnrichedText]) -> list[EnrichedText]:
+    grouped_texts_by_headings: list[EnrichedText] = []
+    first_paragraph = markdown_texts[0]
+    combined_text = ""
+    current_heading = first_paragraph.headings
+    page_num = first_paragraph.page_number
+    elem_type = first_paragraph.type
+    headings_processed: list[str] = []
+    for ind, markdown_text in enumerate(markdown_texts):
+        flat_heading = " ".join([h.title for h in current_heading])
+        current_flat_heading = " ".join([h.title for h in markdown_texts[ind].headings])
```

Comment:
I believe you can keep them as lists and still compare them and check if a heading exists with needing `_find_heading_index()`.

Reply:
The list comparison worked, though I ran into issues with the search in `_find_heading_index` when revising the lookup parameters to search for the heading within the list of processed headings.

Reply:
I'll push a commit to show what I intended.